> I don't know why the pre-commit build failed on Windows. The build itself looks fine but seems to be complaining about slow tests?

Actually, the error is here:

```
FAIL: Flang :: Driver/linker-flags.f90 (218 of 2807)
******************** TEST 'Flang :: Driver/linker-flags.f90' FAILED ********************
Exit Code: 1
Command Output (stdout):
--
# RUN: at line 5
c:\ws\src\build\bin\flang-new.exe -### --target=ppc64le-linux-gnu C:\ws\src\flang\test\Driver/Inputs/hello.f90 2>&1 | c:\ws\src\build\bin\filecheck.exe C:\ws\src\flang\test\Driver\linker-flags.f90 --check-prefixes=CHECK,UNIX,UNIX-F128NONE
# executed command: 'c:\ws\src\build\bin\flang-new.exe' '-###' --target=ppc64le-linux-gnu 'C:\ws\src\flang\test\Driver/Inputs/hello.f90'
# note: command had no output on stdout or stderr
# executed command: 'c:\ws\src\build\bin\filecheck.exe' 'C:\ws\src\flang\test\Driver\linker-flags.f90' --check-prefixes=CHECK,UNIX,UNIX-F128NONE
# note: command had no output on stdout or stderr
# RUN: at line 6
c:\ws\src\build\bin\flang-new.exe -### --target=aarch64-apple-darwin C:\ws\src\flang\test\Driver/Inputs/hello.f90 2>&1 | c:\ws\src\build\bin\filecheck.exe C:\ws\src\flang\test\Driver\linker-flags.f90 --check-prefixes=CHECK,DARWIN,DARWIN-F128NONE
# executed command: 'c:\ws\src\build\bin\flang-new.exe' '-###' --target=aarch64-apple-darwin 'C:\ws\src\flang\test\Driver/Inputs/hello.f90'
# note: command had no output on stdout or stderr
# executed command: 'c:\ws\src\build\bin\filecheck.exe' 'C:\ws\src\flang\test\Driver\linker-flags.f90' --check-prefixes=CHECK,DARWIN,DARWIN-F128NONE
# note: command had no output on stdout or stderr
# RUN: at line 7
c:\ws\src\build\bin\flang-new.exe -### --target=sparc-sun-solaris2.11 C:\ws\src\flang\test\Driver/Inputs/hello.f90 2>&1 | c:\ws\src\build\bin\filecheck.exe C:\ws\src\flang\test\Driver\linker-flags.f90 --check-prefixes=CHECK,UNIX,SOLARIS-F128NONE
# executed command: 'c:\ws\src\build\bin\flang-new.exe' '-###' --target=sparc-sun-solaris2.11 'C:\ws\src\flang\test\Driver/Inputs/hello.f90'
# note: command had no output on stdout or stderr
# executed command: 'c:\ws\src\build\bin\filecheck.exe' 'C:\ws\src\flang\test\Driver\linker-flags.f90' --check-prefixes=CHECK,UNIX,SOLARIS-F128NONE
# note: command had no output on stdout or stderr
# RUN: at line 8
c:\ws\src\build\bin\flang-new.exe -### --target=x86_64-unknown-freebsd C:\ws\src\flang\test\Driver/Inputs/hello.f90 2>&1 | c:\ws\src\build\bin\filecheck.exe C:\ws\src\flang\test\Driver\linker-flags.f90 --check-prefixes=CHECK,UNIX,UNIX-F128NONE
# executed command: 'c:\ws\src\build\bin\flang-new.exe' '-###' --target=x86_64-unknown-freebsd 'C:\ws\src\flang\test\Driver/Inputs/hello.f90'
# note: command had no output on stdout or stderr
# executed command: 'c:\ws\src\build\bin\filecheck.exe' 'C:\ws\src\flang\test\Driver\linker-flags.f90' --check-prefixes=CHECK,UNIX,UNIX-F128NONE
# note: command had no output on stdout or stderr
# RUN: at line 9
c:\ws\src\build\bin\flang-new.exe -### --target=x86_64-unknown-netbsd C:\ws\src\flang\test\Driver/Inputs/hello.f90 2>&1 | c:\ws\src\build\bin\filecheck.exe C:\ws\src\flang\test\Driver\linker-flags.f90 --check-prefixes=CHECK,UNIX,UNIX-F128NONE
# executed command: 'c:\ws\src\build\bin\flang-new.exe' '-###' --target=x86_64-unknown-netbsd 'C:\ws\src\flang\test\Driver/Inputs/hello.f90'
# note: command had no output on stdout or stderr
# executed command: 'c:\ws\src\build\bin\filecheck.exe' 'C:\ws\src\flang\test\Driver\linker-flags.f90' --check-prefixes=CHECK,UNIX,UNIX-F128NONE
# note: command had no output on stdout or stderr
# RUN: at line 10
c:\ws\src\build\bin\flang-new.exe -### --target=x86_64-unknown-openbsd C:\ws\src\flang\test\Driver/Inputs/hello.f90 2>&1 | c:\ws\src\build\bin\filecheck.exe C:\ws\src\flang\test\Driver\linker-flags.f90 --check-prefixes=CHECK,UNIX,UNIX-F128NONE
# executed command: 'c:\ws\src\build\bin\flang-new.exe' '-###' --target=x86_64-unknown-openbsd 'C:\ws\src\flang\test\Driver/Inputs/hello.f90'
# note: command had no output on stdout or stderr
# executed command: 'c:\ws\src\build\bin\filecheck.exe' 'C:\ws\src\flang\test\Driver\linker-flags.f90' --check-prefixes=CHECK,UNIX,UNIX-F128NONE
# note: command had no output on stdout or stderr
# RUN: at line 11
c:\ws\src\build\bin\flang-new.exe -### --target=x86_64-unknown-dragonfly C:\ws\src\flang\test\Driver/Inputs/hello.f90 2>&1 | c:\ws\src\build\bin\filecheck.exe C:\ws\src\flang\test\Driver\linker-flags.f90 --check-prefixes=CHECK,UNIX,UNIX-F128NONE
# executed command: 'c:\ws\src\build\bin\flang-new.exe' '-###' --target=x86_64-unknown-dragonfly 'C:\ws\src\flang\test\Driver/Inputs/hello.f90'
# note: command had no output on stdout or stderr
# executed command: 'c:\ws\src\build\bin\filecheck.exe' 'C:\ws\src\flang\test\Driver\linker-flags.f90' --check-prefixes=CHECK,UNIX,UNIX-F128NONE
# note: command had no output on stdout or stderr
# RUN: at line 12
c:\ws\src\build\bin\flang-new.exe -### --target=x86_64-unknown-haiku C:\ws\src\flang\test\Driver/Inputs/hello.f90 2>&1 | c:\ws\src\build\bin\filecheck.exe C:\ws\src\flang\test\Driver\linker-flags.f90 --check-prefixes=CHECK,HAIKU,HAIKU-F128NONE
# executed command: 'c:\ws\src\build\bin\flang-new.exe' '-###' --target=x86_64-unknown-haiku 'C:\ws\src\flang\test\Driver/Inputs/hello.f90'
# note: command had no output on stdout or stderr
# executed command: 'c:\ws\src\build\bin\filecheck.exe' 'C:\ws\src\flang\test\Driver\linker-flags.f90' --check-prefixes=CHECK,HAIKU,HAIKU-F128NONE
# note: command had no output on stdout or stderr
# RUN: at line 13
c:\ws\src\build\bin\flang-new.exe -### --target=x86_64-windows-gnu C:\ws\src\flang\test\Driver/Inputs/hello.f90 2>&1 | c:\ws\src\build\bin\filecheck.exe C:\ws\src\flang\test\Driver\linker-flags.f90 --check-prefixes=CHECK,MINGW,MINGW-F128NONE
# executed command: 'c:\ws\src\build\bin\flang-new.exe' '-###' --target=x86_64-windows-gnu 'C:\ws\src\flang\test\Driver/Inputs/hello.f90'
# note: command had no output on stdout or stderr
# executed command: 'c:\ws\src\build\bin\filecheck.exe' 'C:\ws\src\flang\test\Driver\linker-flags.f90' --check-prefixes=CHECK,MINGW,MINGW-F128NONE
# note: command had no output on stdout or stderr
# RUN: at line 14
c:\ws\src\build\bin\flang-new.exe -### -rtlib=compiler-rt --target=aarch64-linux-gnu C:\ws\src\flang\test\Driver/Inputs/hello.f90 2>&1 | c:\ws\src\build\bin\filecheck.exe C:\ws\src\flang\test\Driver\linker-flags.f90 --check-prefixes=CHECK,UNIX,COMPILER-RT
# executed command: 'c:\ws\src\build\bin\flang-new.exe' '-###' -rtlib=compiler-rt --target=aarch64-linux-gnu 'C:\ws\src\flang\test\Driver/Inputs/hello.f90'
# note: command had no output on stdout or stderr
# executed command: 'c:\ws\src\build\bin\filecheck.exe' 'C:\ws\src\flang\test\Driver\linker-flags.f90' --check-prefixes=CHECK,UNIX,COMPILER-RT
# .---command stderr------------
# | C:\ws\src\flang\test\Driver\linker-flags.f90:31:16: error: COMPILER-RT: expected string not found in input
# | ! COMPILER-RT: "{{.*}}/clang_rt.crtbegin.o"
# |                ^
# | <stdin>:7:27: note: scanning from here
# |  "c:\\llvm-mingw\\bin\\ld" "-EL" "--hash-style=gnu" "--eh-frame-hdr" "-m" "aarch64linux" "-pie" "-dynamic-linker" "/lib/ld-linux-aarch64.so.1" "-o" "a.out" "Scrt1.o" "crti.o" "crtbeginS.o" "C:\\Users\\ContainerAdministrator\\AppData\\Local\\Temp\\lit-tmp-9wgczl7q\\hello-1c57c7.o" "-LC:\\ws\\src\\build\\lib" "-lFortranRuntime" "-lFortranDecimal" "-lm" "C:\\ws\\src\\build\\lib\\clang\\19\\lib\\aarch64-unknown-linux-gnu\\libclang_rt.builtins.a" "-lc" "C:\\ws\\src\\build\\lib\\clang\\19\\lib\\aarch64-unknown-linux-gnu\\libclang_rt.builtins.a" "crtendS.o" "crtn.o"
# |                           ^
# | <stdin>:7:163: note: possible intended match here
# |  "c:\\llvm-mingw\\bin\\ld" "-EL" "--hash-style=gnu" "--eh-frame-hdr" "-m" "aarch64linux" "-pie" "-dynamic-linker" "/lib/ld-linux-aarch64.so.1" "-o" "a.out" "Scrt1.o" "crti.o" "crtbeginS.o" "C:\\Users\\ContainerAdministrator\\AppData\\Local\\Temp\\lit-tmp-9wgczl7q\\hello-1c57c7.o" "-LC:\\ws\\src\\build\\lib" "-lFortranRuntime" "-lFortranDecimal" "-lm" "C:\\ws\\src\\build\\lib\\clang\\19\\lib\\aarch64-unknown-linux-gnu\\libclang_rt.builtins.a" "-lc" "C:\\ws\\src\\build\\lib\\clang\\19\\lib\\aarch64-unknown-linux-gnu\\libclang_rt.builtins.a" "crtendS.o" "crtn.o"
# |                                                                                                                                                                   ^
# |
# | Input file: <stdin>
# | Check file: C:\ws\src\flang\test\Driver\linker-flags.f90
# |
# | -dump-input=help explains the following input dump.
# |
# | Input was:
# | <<<<<<
# |             1: flang-new version 19.0.0git (https://github.com/llvm/llvm-project.git 66e13f92a0680742b552fabde25df7752f8510c9)
# |             2: Target: aarch64-unknown-linux-gnu
# |             3: Thread model: posix
# |             4: InstalledDir: C:\ws\src\build\bin
# |             5: Build config: +assertions
# |             6:  "C:\\ws\\src\\build\\bin\\flang-new" "-fc1" "-triple" "aarch64-unknown-linux-gnu" "-emit-obj" "-mrelocation-model" "pic" "-pic-level" "2" "-pic-is-pie" "-target-cpu" "generic" "-target-feature" "+outline-atomics" "-target-feature" "+v8a" "-target-feature" "+fp-armv8" "-target-feature" "+neon" "-resource-dir" "C:\\ws\\src\\build\\lib\\clang\\19" "-mframe-pointer=non-leaf" "-o" "C:\\Users\\ContainerAdministrator\\AppData\\Local\\Temp\\lit-tmp-9wgczl7q\\hello-1c57c7.o" "-x" "f95-cpp-input" "C:\\ws\\src\\flang\\test\\Driver/Inputs/hello.f90"
# |             7:  "c:\\llvm-mingw\\bin\\ld" "-EL" "--hash-style=gnu" "--eh-frame-hdr" "-m" "aarch64linux" "-pie" "-dynamic-linker" "/lib/ld-linux-aarch64.so.1" "-o" "a.out" "Scrt1.o" "crti.o" "crtbeginS.o" "C:\\Users\\ContainerAdministrator\\AppData\\Local\\Temp\\lit-tmp-9wgczl7q\\hello-1c57c7.o" "-LC:\\ws\\src\\build\\lib" "-lFortranRuntime" "-lFortranDecimal" "-lm" "C:\\ws\\src\\build\\lib\\clang\\19\\lib\\aarch64-unknown-linux-gnu\\libclang_rt.builtins.a" "-lc" "C:\\ws\\src\\build\\lib\\clang\\19\\lib\\aarch64-unknown-linux-gnu\\libclang_rt.builtins.a" "crtendS.o" "crtn.o"
# | check:31'0                               X~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ error: no match found
# | check:31'1                                                                                                                                                                       ?                                                                                                                                                                                                                                                                                                                                                                                                                    possible intended match
# | >>>>>>
# `-----------------------------
# error: command failed with exit status: 1
--
```

--- a/flang/test/Driver/linker-flags.f90
+++ b/flang/test/Driver/linker-flags.f90
@@ -34,7 +34,7 @@
 ! UNIX-F128LIBQUADMATH-SAME: "-lFortranFloat128Math" "--as-needed" "-lquadmath" "--no-as-needed"
 ! SOLARIS-F128LIBQUADMATH-SAME: "-lFortranFloat128Math" "-z" "ignore" "-lquadmath" "-z" "record"
 ! UNIX-SAME: "-lFortranRuntime" "-lFortranDecimal" "-lm"
-! COMPILER-RT: "{{.*}}/libclang_rt.builtins.a"
+! COMPILER-RT: "{{.*}}{{\\|/}}libclang_rt.builtins.a"
 
 ! DARWIN-LABEL:  "{{.*}}ld{{(\.exe)?}}"
 ! DARWIN-SAME: "[[object_file]]"

--- a/flang/test/Driver/linker-flags.f90
+++ b/flang/test/Driver/linker-flags.f90
@@ -28,14 +28,13 @@
 !       suffix. Clang's driver will try to resolve the path to the ld
 !       executable and may find the GNU linker from MinGW or Cygwin.
 ! UNIX-LABEL:  "{{.*}}ld{{(\.exe)?}}"
-! COMPILER-RT: "{{.*}}/clang_rt.crtbegin.o" 
 ! UNIX-SAME: "[[object_file]]"
 ! UNIX-F128NONE-NOT: FortranFloat128Math
 ! SOLARIS-F128NONE-NOT: FortranFloat128Math
 ! UNIX-F128LIBQUADMATH-SAME: "-lFortranFloat128Math" "--as-needed" "-lquadmath" "--no-as-needed"
 ! SOLARIS-F128LIBQUADMATH-SAME: "-lFortranFloat128Math" "-z" "ignore" "-lquadmath" "-z" "record"
 ! UNIX-SAME: "-lFortranRuntime" "-lFortranDecimal" "-lm"
-! COMPILER-RT: "{{.*}}/libclang_rt.builtins.a" "{{.*}}/clang_rt.crtend.o"
+! COMPILER-RT: "{{.*}}/libclang_rt.builtins.a"
 
 ! DARWIN-LABEL:  "{{.*}}ld{{(\.exe)?}}"
 ! DARWIN-SAME: "[[object_file]]"

